ompi/mca/coll/acoll/coll_acoll_allgather.c

```diff
@@ -0,0 +1,484 @@
+/* -*- Mode: C; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (C) 2024, Advanced Micro Devices, Inc. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+
+#include "mpi.h"
+#include "ompi/constants.h"
+#include "ompi/datatype/ompi_datatype.h"
+#include "ompi/mca/coll/base/coll_base_functions.h"
+#include "ompi/mca/coll/base/coll_base_util.h"
+#include "ompi/mca/coll/base/coll_tags.h"
+#include "ompi/mca/coll/coll.h"
+#include "ompi/mca/pml/pml.h"
+#include "opal/util/bit_ops.h"
+#include "coll_acoll.h"
+#include "coll_acoll_utils.h"
+
+/*
+ * coll_allgather_decision_fixed
+ *
+ * Function:    Choose optimal allgather algorithm
+ *
+ * Description: Based on no. of processes and message size, chooses whether
+ *              or not to use subgroups. If subgroup based algorithm is not,
+ *              chosen, further decides if [ring|lin] allgather is to be used.
+ *
+ */
+static inline void coll_allgather_decision_fixed(int size, size_t total_dsize, int sg_size,
```

Comment:
can you shed some lights on how to choose which methods for other intel/amd achitectures?  you might also want some utility to let the user to adjust the decisions according for other systems.

Reply:
Our testing has been mostly focused on Zen architectures, we will soon test on other architectures. We do not have the utility/config option to override decisions, we will plan to add it.